### ISSUE TITLE
nbf / iat clock skew not taken into account

### ISSUE THREAD

**Issue by antgel** [blake-simpson/expo-jwt#7]
Hi, thanks for such a useful library.

I'm encoding JWT tokens on my server using [flask-jwt-extended](https://flask-jwt-extended.readthedocs.io), and decoding them on my device using expo-jwt.

My app is very simple now, something like:

```javascript
  loginAsync = async () => {
    const res = await Api.post('/users/login/', {
      username: this.state.username,
    });
    await AsyncStorage.setItem('accessToken', res.body.access_token);
    await AsyncStorage.setItem('refreshToken', res.body.refresh_token);
    Api.setAccessToken(res.body.access_token);

    const decodedJwt = JWT.decode(res.body.access_token, Api.jwt_key);
    const userId = decodedJwt.identity;
    await AsyncStorage.setItem('userId', String(userId));
  };
```

My `JWT.decode` occasionally (quite a lot of the time) fails. I checked in `verifier.js`, and noticed:
```javascript
      if (nbf > this.time) {
        throw new Errors.ImmatureSignature();
      }
```

At first, I assumed my server was out, as ntp wasn't running. But it wasn't that. Even after installing ntp and giving it ample time to sync, for whatever reason, `this.time` would sometimes be a second *behind* `nbf`.

I don't know much about clock sync or skew on Android, but I have verified that my device epoch seconds lag ever so slightly behind my server, on a regular basis. "Automatic date & time" is set in the Android system settings.

Given this situation, and that in the field, apparently I should expect some level of inaccuracy even from "synchronized" devices, what do you suggest? Would it be possible to implement a skew parameter? This is in line with [the RFC:](https://tools.ietf.org/html/rfc7519) 

> Implementers MAY provide for some small leeway, usually no more than a few minutes, to account for clock skew.

**Comment by blake-simpson** [comment]
Hi,

Interesting point. I had not considered that before. I think adding a `skew` parameter with a millisecond value that could be added to the decode options would make sense.

If you have made changes already, feel free open a PR. Otherwise I will aim to release a new package with this option included soon.

**Comment by blake-simpson** [comment]
Hi @antgel,

While I was investigating this I found the skew changes would be quite straight forward, so I've gone ahead and added them now.

You can see the new API in the README but basically you can pass a number of seconds as a `timeSkew` option to the `decode` function, like so:

```js
// Allow verification of tokens that include "exp", "nbf", or "iat" claims within an additional 30 seconds of the system time.
JWT.decode(token, key, { timeSkew: 30 });
```

I've released that to NPM as version 1.2.0, so please try it out and I hope it resolves your issue :) If not, let me know and we can see how to fix it.

**Comment by antgel** [comment]
@blake-simpson Outstanding, thanks so much for the insight and the patch. I'll implement the upgrade ASAP and report back if there's still an issue. :)
